### PR TITLE
Add file status on API

### LINKED ISSUE #4313: Webhook POST data doesn't pass added/modified/removed info in commits section
- Gitea version (or commit ref): 1.4.2
- Git version: 2.13
- Operating system: Linux
- Database (use `[x]`): 
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [X] SQLite
- Can you reproduce the bug at https://try.gitea.io: (URL is down at the time of issue)
  - [X] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description
I created a web hook and pushed a change (Tried with both content types json and xml)
When a looked at payload sent by gitea in commit section 
"commits": [
    {
      "id": "bffeb74224043ba2feb48d137756c8a9331c449a",
      "message": "Webhooks Yay!",
      "url": "http://localhost:3000/gitea/webhooks/commit/bffeb74224043ba2feb48d137756c8a9331c449a",
      "author": {
        "name": "Gitea",
        "email": "someone@gitea.io",
        "username": "gitea"
      },
      "committer": {
        "name": "Gitea",
        "email": "someone@gitea.io",
        "username": "gitea"
      },
      "timestamp": "2017-03-13T13:52:11-04:00"
    }
  ]

there is no added/modified/removed info

"commits": [
    {
      "id": "bffeb74224043ba2feb48d137756c8a9331c449a",
      "message": "Webhooks Yay!",
      "url": "http://localhost:3000/gitea/webhooks/commit/bffeb74224043ba2feb48d137756c8a9331c449a",
      "author": {
        "name": "Gitea",
        "email": "someone@gitea.io",
        "username": "gitea"
      },
      "committer": {
        "name": "Gitea",
        "email": "someone@gitea.io",
        "username": "gitea"
      },
      "timestamp": "2017-03-13T13:52:11-04:00"
    }
    "added": ...

  ]

### PR DESCRIPTION
**Pull request by lunny**
Will replace #5607 and fix #4313

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
I don't see anything in diff that should have changed commits in tests or am I missing something?

**Comment by sapk** [comment]
Tests are failing since you check the repo content in tests now and user2/repo2.git is not initialized.

**Comment by lunny** [comment]
@lafriks get file status will read git repository, and previous tests will read a non-exist repository, so I change it to an exist repository. I have copied all repositories from `integration/gitea-repositories-meta` in a previous PR that means all unit tests could really read/write git repositories.

@sapk tests fixed.

File: models/action.go [hunk 1]
```diff
@@ -431,10 +438,13 @@ func (pc *PushCommits) ToAPIPayloadCommits(repoLink string) []*api.PayloadCommit
 				Email:    commit.CommitterEmail,
 				UserName: committerUsername,
 			},
+			Added:     fileStatus.Added,
```
Comment (techknowlogick) [inline models/action.go:441]: Can you add tests to see if `added`, `removed` and `modified` show up with correct information?

Comment (lunny) [reply models/action.go:441]: done.

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**